src/sideEffect/saga/referenceFetch.js

```diff
@@ -11,34 +11,33 @@ import { crudGetMany } from '../../actions/dataActions';
  *   authors: { 23: true, 47: true, 78: true },
  * }
  */
-let ids = {};
+const ids = {};
+const tasks = {};
 
 // see http://yelouafi.github.io/redux-saga/docs/recipes/index.html#debouncing
-function *fetchReference(resource) {
+function* fetchReference(resource) {
     // combined with cancel(), this debounces the calls
     yield call(delay, 50);
-    yield Object.keys(ids).map(reference =>
```

Comment:
There was a logic error here: if we decide to fetch one resource, we shouldn't iterate on the keys of `ids`. We know the resource to fetch.